pkg/messaging/v1/util.go

```diff
@@ -154,7 +154,7 @@ func InternalMetadataToHTTPHeader(internalMD DaprInternalMetadata, setHeader fun
 		if len(listVal.Values) == 0 || strings.HasSuffix(k, gRPCBinaryMetadataSuffix) || k == ContentTypeHeader || isTraceCorrleationHeaderKey(k) {
 			continue
 		}
-		setHeader(reservedGRPCMetadataToDaprPrefixHeader(k), listVal.Values[0])
```

Comment:
Need some info on why we were using just the first element?

cc: @youngbupark

Reply:
this is by-design. we only take the first value when we convert internal metadata to http header.

Reply:
Thanks for explaining, reverted this change.